docs/client-libraries-java.md

```diff
@@ -29,12 +29,12 @@ The following table outlines the API packages and reference docs for Pulsar Java
 
 Package | Description | Maven Artifact
 :-------|:------------|:--------------
-[`org.apache.pulsar.client.api`](/api/client) | [Java client API](/api/client/) | [org.apache.pulsar:pulsar-client:@pulsar:version@](http://search.maven.org/#artifactdetails%7Corg.apache.pulsar%7Cpulsar-client%7C@pulsar:version@%7Cjar)
+[`org.apache.pulsar.client.api`](/api/client) | [Java client API](client-api-overview.md) | [org.apache.pulsar:pulsar-client:@pulsar:version@](http://search.maven.org/#artifactdetails%7Corg.apache.pulsar%7Cpulsar-client%7C@pulsar:version@%7Cjar)
```

Comment:
I don't think `client-api-overview` is the correct page to be linked here. Actually, I'd suggest to merge that page into clent-libraries-overview.

Reply:
I agree it doesn't make 100% sense. The initial motivation to change the link is because `/api/client` has been linked in the first column. I think it would be redundant to provide the reference link again.
@Anonymitaet WDYT?

Reply:
Since now we only have 1-page content for https://pulsar.apache.org/docs/next/client-api-overview/, it can go to `Reference > API > Client` with API reference doc links for three languages (Java, Python, C++) since they are more likely to belong to "Reference" topic rather than "Concept" topic (Client Libraries > Overview).

Reply:
To me, the paragraph in https://pulsar.apache.org/docs/next/client-api-overview/ looks more like a concept introduction, where we provide descriptions and an illustration for explanations.
I intend to relocate it to 'Pulsar API Overview' since it's the most related topic. @Anonymitaet - I wonder whether it conflicts with your plan on improving/expanding API docs. If yes, I think `Client library Overview` can be an alternative.

Reply:
Feel free to make your desired changes.